lib/compat/strnlen.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2018 Balabit
+ * Copyright (c) 2018 Kokan
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "compat/string.h"
+
+#if !SYSLOG_NG_HAVE_STRNLEN
+
+#include <stdint.h>
```

Comment:
according to man 3 memchr, memchr is included from string.h, which is included from compat/string.h. Do we need this include?

Reply:
This is not included because of the `memchr`, that is guaranteed to be included for the reason you stated above.
And I totally meant `stddef.h` not `stdint.h` :), that is needed for `size_t`. I guess the `stddef.h` is included at some place either way.